lib/es6-promise/polyfill.js

```diff
@@ -2,7 +2,20 @@
 import Promise from './promise';
 
 export default function polyfill() {
-  var local = (typeof global !== 'undefined') ? global : self;
+  var local;
+
+  if (typeof global !== 'undefined') {
+      local = global;
+  } else if (typeof self !== 'undefined') {
+      local = self;
+  } else {
+      try {
+          local = Function('return this')();
```

Comment:
Yeah, this is the same as `eval`, but as the document of CSP suggests, this only throws EvalError(or Error) and doesn't invalidate the entire script.

To check that behaviour, I created a sample page below:

http://csp-exp.herokuapp.com/
(source code: https://github.com/kt3k/csp-exp )

In the above page, the calls of `Function` and `eval` throw errors but the other part of script works normally. (I checked in Chrome, FF, Safari)